### PR TITLE
Add shlex utility

### PR DESCRIPTION
**Pull request by elharo**
@chanseokoh @briandealwis 

Please suggest bugs or suggestion for additional features in the form of test cases not currently handled. E.g.

foo='bar baz'
or
foo="bar\"baz"

I'm not sure how far we need to go with this for version 1.

### CODE REVIEW

File: src/main/java/com/google/cloud/tools/shell/CommandLine.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+  /**
+   * Split a full command line into an array of individual arguments,
+   * similar to the shlex function in Python according to POSIX rules. 
+   * All input characters are preserved except for separating whitespace. 
+   * This function tokenizes the input string, but does not attempt to parse it.
+   * For example, it recognizes that a space escaped with a backslash does not
+   * start a new token, but it does not replace the "\ " with a single space. 
+   * 
+   * @param line the input line
```
Comment (chanseokoh) [inline src/main/java/com/google/cloud/tools/shell/CommandLine.java]: Add "cannot be null"?

File: src/main/java/com/google/cloud/tools/shell/CommandLine.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+   * @return a non-null but possibly empty array of arguments
+   */
+  public static String[] split(String line) {    
+    char quote = '"';
```
Comment (chanseokoh) [inline src/main/java/com/google/cloud/tools/shell/CommandLine.java]: I think initializing this to a particular quote adds confusion. The value is not used and will just be overwritten, right?

Comment (elharo) [inline src/main/java/com/google/cloud/tools/shell/CommandLine.java]: It needs to be initialized to something though:

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.5.1:compile (default-compile) on project appengine-plugins-core: Compilation failure
[ERROR] /Users/elharo/appengine-plugins-core/src/main/java/com/google/cloud/tools/shell/CommandLine.java:[52,35] variable quote might not have been initialized
[ERROR] -> [Help 1]
[ERROR]

File: src/main/java/com/google/cloud/tools/shell/CommandLine.java (hunk 1 above)
Comment (elharo) [inline src/main/java/com/google/cloud/tools/shell/CommandLine.java]: added @throws

**Comment by elharo** [comment]
Ping @chanseokoh

File: src/test/java/com/google/cloud/tools/shell/CommandLineTest.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
+/* Copyright 2017 Google Inc.
```
Comment (chanseokoh) [inline src/test/java/com/google/cloud/tools/shell/CommandLineTest.java]: Ditto.

**Comment by briandealwis** [comment]
The original motivation for this work was https://github.com/GoogleCloudPlatform/google-cloud-eclipse/pull/1139. As there is an alternative in Eclipse, is this necessary?

**Comment by loosebazooka** [comment]
reopen if you still intend on doing this.

**Pull request closed without merging**